### PR TITLE
Merge AHR-82/list-standard-oneline to AHR-77/organisms

### PR DESCRIPTION
**Pull request by mohamed-elshafey-halan**

### CODE REVIEW

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -9,6 +9,9 @@
         android:roundIcon="@mipmap/ic_launcher_round"
         android:supportsRtl="true"
         android:theme="@style/Theme.MyApplication">
+        <activity
+            android:name=".organisms.ListStandardActivity"
+            android:exported="true" />
```
Comment (OBeshary) [inline app/src/main/AndroidManifest.xml:14]: why using `android:exported="true"` tag ?

File: app/src/main/java/com/example/myapplication/MainActivity.kt [hunk 2]
```diff
@@ -11,5 +20,33 @@ class MainActivity : AppCompatActivity() {
...
+        val primaryButton = findViewById<PrimaryButton>(R.id.primaryButton)
+//        primaryButton.setOnClickListener {
+////            primaryButton.isLoading = primaryButton.isLoading.not()
+//            if (primaryButton.buttonState == PrimaryButton.ButtonState.DANGER) {
+//                primaryButton.buttonState = PrimaryButton.ButtonState.NORMAL
+//                primaryButton.isLoading = true
+//            } else {
+//                primaryButton.buttonState = PrimaryButton.ButtonState.DANGER
+//                primaryButton.isLoading = false
+//            }
```
Comment (OBeshary) [inline app/src/main/java/com/example/myapplication/MainActivity.kt:35]: there is no need to push commented code.

File: app/src/main/java/com/example/myapplication/organisms/ListStandardActivity.kt [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+            itemList.showLeading = isChecked
+        }
+
+        showIconSwitch.setOnCheckedChangeListener { buttonView, isChecked ->
+            if (isChecked) {
+                itemList.icon = ContextCompat.getDrawable(this, R.drawable.ic_wallet)
+            } else {
+                itemList.icon = null
+            }
+
```
Comment (OBeshary) [inline app/src/main/java/com/example/myapplication/organisms/ListStandardActivity.kt:34]: remove this line

File: app/src/main/java/com/example/myapplication/organisms/ListStandardActivity.kt [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+import androidx.core.content.ContextCompat
+import com.example.halan_android_design_system.organisms.item_list_standard_oneline.ItemListStandardOneLineType
+import com.example.myapplication.R
+import kotlinx.android.synthetic.main.activity_list_standard.*
+
+class ListStandardActivity : AppCompatActivity() {
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_list_standard)
+
```
Comment (OBeshary) [inline app/src/main/java/com/example/myapplication/organisms/ListStandardActivity.kt:17]: remove this line

File: app/src/main/java/com/example/myapplication/organisms/ListStandardActivity.kt [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+        setTypeSpinner()
+
+        showLeadingSwitch.isChecked = itemList.showLeading
+        showIconSwitch.isChecked = itemList.icon != null
+        showDescriptionSwitch.isChecked = !itemList.descriptionText.isNullOrBlank()
+
+        showLeadingSwitch.setOnCheckedChangeListener { buttonView, isChecked ->
+            itemList.showLeading = isChecked
```
Comment (OBeshary) [inline app/src/main/java/com/example/myapplication/organisms/ListStandardActivity.kt:25]: `buttonView` is not used should be renamed to `_`

File: app/src/main/java/com/example/myapplication/organisms/ListStandardActivity.kt [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+        }
+
+        showIconSwitch.setOnCheckedChangeListener { buttonView, isChecked ->
+            if (isChecked) {
+                itemList.icon = ContextCompat.getDrawable(this, R.drawable.ic_wallet)
+            } else {
+                itemList.icon = null
```
Comment (OBeshary) [inline app/src/main/java/com/example/myapplication/organisms/ListStandardActivity.kt:32]: this condition could be simplified to `itemList.icon = if (isChecked) ContextCompat.getDrawable(this, R.drawable.ic_wallet) else null`

File: app/src/main/java/com/example/myapplication/organisms/ListStandardActivity.kt [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+            }
+
+        }
+
+        showDescriptionSwitch.setOnCheckedChangeListener { _, isChecked ->
+            if (isChecked) {
+                itemList.descriptionText = resources.getString(R.string.description_list_standard_oneline)
+            } else {
+                itemList.descriptionText = null
+            }
```
Comment (OBeshary) [inline app/src/main/java/com/example/myapplication/organisms/ListStandardActivity.kt:42]: this condition could be simplified to `itemList. descriptionText = if (isChecked) resources.getString(R.string.description_list_standard_oneline) else null`

File: app/src/main/res/layout/activity_list_standard.xml [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+    tools:context=".organisms.ListStandardActivity">
+
+
+    <androidx.appcompat.widget.AppCompatSpinner
+        android:id="@+id/typeSpinner"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:layout_marginTop="16dp"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
```
Comment (OBeshary) [inline app/src/main/res/layout/activity_list_standard.xml:16]: i think you can remove both fields as you are using `android:layout_width="match_parent"
`

File: app/src/main/res/layout/activity_list_standard.xml [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+        android:id="@+id/itemList"
+        android:layout_width="0dp"
+        android:layout_height="wrap_content"
+        android:icon="@drawable/ic_wallet"
+        app:description="@string/description_list_standard_oneline"
+        app:iconDisabled="@drawable/ic_wallet_disabled"
+        app:itemListStandardType="primary"
+        app:layout_constraintBottom_toBottomOf="parent"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
```
Comment (OBeshary) [inline app/src/main/res/layout/activity_list_standard.xml:60]: use `android:layout_width="match_parent"`

File: halan-android-design-system/src/main/java/com/example/halan_android_design_system/atom/PrimaryButton.kt [hunk 10]
```diff
@@ -0,0 +1,176 @@
...
+
+                insetTop = 0
+                insetBottom = 0
+                cornerRadius = resources.getDimension(buttonSize.cornerRadiusDimen).toInt()
+                TextViewCompat.setTextAppearance(this@PrimaryButton, buttonSize.styleRes)
+            } finally {
+                recycle()
+            }
+        }
+        //TODO for testing purpose only
```
Comment (OBeshary) [inline halan-android-design-system/src/main/java/com/example/halan_android_design_system/atom/PrimaryButton.kt:102]: remove this comment

File: halan-android-design-system/src/main/java/com/example/halan_android_design_system/atom/PrimaryButton.kt [hunk 11]
```diff
@@ -0,0 +1,176 @@
...
+            invalidate()
+        } else {
+            iconPadding = resources.getDimension(R.dimen.default_icon_padding).toInt()
+        }
+
+        icon.setTint(currentTextColor)
+    }
+
+    private fun saveDefaultText() {
+        if (text?.isNotEmpty() == true)
```
Comment (OBeshary) [inline halan-android-design-system/src/main/java/com/example/halan_android_design_system/atom/PrimaryButton.kt:151]: use `IsNullOrEmpty`

File: halan-android-design-system/src/main/java/com/example/halan_android_design_system/organisms/item_list_standard_oneline/ItemListStandardOneLine.kt [hunk 12]
```diff
@@ -0,0 +1,115 @@
...
+            else
+                iconImageView.setImageDrawable(icon)
+        } else {
+            iconImageView.setImageDrawable(iconDisabled)
+        }
+
+        titleTextView.text = titleText
+
+        descriptionTextView.text = descriptionText
+        descriptionTextView.visibility = if (descriptionText.isNullOrEmpty()) View.GONE else View.VISIBLE
```
Comment (OBeshary) [inline halan-android-design-system/src/main/java/com/example/halan_android_design_system/organisms/item_list_standard_oneline/ItemListStandardOneLine.kt:106]: would be prefer if you statically imported `View.GONE , View.VISIBLE & View.IN_VISIBLE`

File: halan-android-design-system/src/main/java/com/example/halan_android_design_system/atom/PrimaryButton.kt [hunk 13]
```diff
@@ -0,0 +1,176 @@
...
+            defaultIconPadding = iconPadding
+    }
+
+    private fun setTheme(backgroundColor: Int, contentColor: Int) {
+        setBackgroundColor(ContextCompat.getColor(context, backgroundColor))
+        setTextColor(ContextCompat.getColor(context, contentColor))
+    }
+
+    inline fun <reified T : Enum<T>> TypedArray.getEnum(index: Int, default: T) =
+        getInt(index, -1).let {
```
Comment (OBeshary) [inline halan-android-design-system/src/main/java/com/example/halan_android_design_system/atom/PrimaryButton.kt:172]: we already have `TypedArrayExtensions.kt` file

File: halan-android-design-system/src/main/res/values/dimens.xml [hunk 14]
```diff
@@ -0,0 +1,18 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+
+    <dimen name="button_large">56dp</dimen>
+    <dimen name="button_medium">48dp</dimen>
+    <dimen name="button_small">40dp</dimen>
+    <dimen name="button_x_small">32dp</dimen>
+
```
Comment (OBeshary) [inline halan-android-design-system/src/main/res/values/dimens.xml:8]: remove this line

File: app/src/main/java/com/example/myapplication/organisms/ListStandardActivity.kt (hunk 3 above)
Comment (mohamed-elshafey-halan) [reply app/src/main/java/com/example/myapplication/organisms/ListStandardActivity.kt:34]: Why?

File: halan-android-design-system/src/main/java/com/example/halan_android_design_system/atom/PrimaryButton.kt (hunk 13 above)
Comment (mohamed-elshafey-halan) [reply halan-android-design-system/src/main/java/com/example/halan_android_design_system/atom/PrimaryButton.kt]: Merged now 👍

**Review by noureldeen-abouelkassem — APPROVED** [review]

**Review by OBeshary — APPROVED** [review]

**Review by OBeshary — APPROVED** [review]

**Review by noureldeen-abouelkassem — APPROVED** [review]

**Pull request merged**